### PR TITLE
`Adaptive learning`: Fix score display for competency details view

### LINKED ISSUE #7600: Tasks incorrectly marked as not completed in "Competencies"
### Describe the bug

Bug was reported in this [Zulip-post](https://zulip.in.tum.de/#narrow/stream/1241-PGdP-Organisation/topic/Aufgaben.20f.C3.A4lschlicherweise.20als.20nicht.20erledigt.20markiert/near/1278726)

The student reported that he submitted the task before the submission date. However, under the tab "Competencies" it said that he missed the submission date. 

### To Reproduce

1. Create an exercise with a competency where the due date is in the future
2. Create a submission to the exercise
3. Go to "Competencies" 
4. Now you should see the competency with the exercise
5. On the right side above the Due Date timer, it should incorrectly say "You have missed the due date"

### Expected behavior

The task should be marked as completed.

### Screenshots

image
image

### Which version of Artemis are you seeing the problem on?

6.6.6

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Resolves #7600 

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Instructor
- 1 Programming Exercise with a submission. The deadline has to be in the future.

1. Log in as a Student
2. Go to "Competencies" 
3. Now you should see the competency with the exercise, navigate to it
4. On the right side above the Due Date timer, it should show your score.
5. Wait until the deadline has passed (or set it in the past)
6. Go to "Competencies"
7. Now you should see the competency with the exercise, navigate to it
8. On the right side above the Due Date timer, it should show your score.

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by JanaNF — DISMISSED** [review]
Manual tested on legacy_ts2. Worked as expected.

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
Do we really need all submissions and all results? Consider if only the latest submission / result is also fine.

Also note that some results should ne be sent to the user, e.g. if a manual assessment ist done but the assessment due date is not over yet. Simply loading all results here is not correct.

**Review by lennart-keller — APPROVED** [review]
Code lgtm :)

File: src/main/java/de/tum/in/www1/artemis/repository/CompetencyRepository.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public interface CompetencyRepository extends JpaRepository<Competency, Long> {
     Optional<Competency> findByIdWithLectureUnits(@Param("competencyId") long competencyId);
 
     /**
-     * Fetches a competency with all linked exercises, lecture units, the associated progress, and completion of the specified user.
+     * Fetches a competency with all linked exercises, lecture units, the associated participations, progress, and completion of the specified user.
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/CompetencyRepository.java:53]:
```suggestion
     * Fetches a competency with all linked exercises, lecture units, the associated progress, and completion of the specified user.
```

**Review by lennart-keller — APPROVED** [review]
Reapprove code

**Comment by Ceiridge** [comment]
It's not possible to add a programming exercise to a competency, but only text units for example:
image

**Review by Gusti2010 — APPROVED** [review]
Tested on TS11_Legacy. Works as expected!

image

**Review by georgyia — APPROVED** [review]
tested manually on ts2, works as expected

**Review by LenaKahle — APPROVED** [review]
Approve, tested on legacy ts11.

**Pull request merged**